packages/flutter_tools/lib/src/android/adb.dart

```diff
@@ -112,6 +118,22 @@ class Adb {
     return controller.stream;
   }
 
+  Future _populateDeviceNames(List<AdbDevice> devices) async {
+    for (AdbDevice device in devices) {
+      if (device.modelID == null) {
+        if (_idToNameCache[device.id] == null)
+          await _populateDeviceCache();
+        device.modelID = _idToNameCache[device.id];
```

Comment:
Ah, I see.  That wasn't really clear from reading the code, maybe a comment would help?